File: src/engine/entities/character.ts
import { Gender } from '@engine/entities/gender'
import { CharacterSkill } from '@engine/entities/character-skill'

export type Name = string
export type Mood = number
export type Health = number
export type Fame = number

export type Character = {
    name: Name
    birthday: Date
    gender: Gender
    mood: Mood
    health: Health
    fame: Fame
    skills: ReadonlyArray<CharacterSkill>
}

File: crates/app/src/entities/character.rs
use chrono::NaiveDate;
use serde::Deserialize;

use super::deserializers;
use super::SkillWithLevel;

#[derive(Deserialize)]
pub enum Gender {
    Male,
    Female,
    Other,
}

#[derive(Deserialize)]
pub struct Character {
    name: String,
    #[serde(with = "deserializers::naivedate")]
    birthday: NaiveDate,
    gender: Gender,
    mood: i8,
    health: i8,
    fame: i8,
    skills: Vec<SkillWithLevel>,
}
